tests/e2e-inst-signatures/e2e-process_execute_failed.go

```diff
@@ -0,0 +1,107 @@
+package main
+
+import (
+	"fmt"
+	"strings"
+	"sync"
+
+	helpers2 "github.com/aquasecurity/libbpfgo/helpers"
+
+	"github.com/aquasecurity/tracee/signatures/helpers"
+	"github.com/aquasecurity/tracee/types/detect"
+	"github.com/aquasecurity/tracee/types/protocol"
+	"github.com/aquasecurity/tracee/types/trace"
+)
+
+type e2eProcessExecuteFailed struct {
+	cb                    detect.SignatureHandler
+	osInfo                *helpers2.OSInfo
+	markUnsupportedKernel sync.Once
+}
+
+func (sig *e2eProcessExecuteFailed) Init(ctx detect.SignatureContext) error {
+	sig.cb = ctx.Callback
+	var err error
+	sig.osInfo, err = helpers2.GetOSInfo()
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (sig *e2eProcessExecuteFailed) GetMetadata() (detect.SignatureMetadata, error) {
+	return detect.SignatureMetadata{
+		ID:          "PROCESS_EXECUTE_FAILED",
+		EventName:   "PROCESS_EXECUTE_FAILED",
+		Version:     "0.1.0",
+		Name:        "Process Execute Failed Test",
+		Description: "Instrumentation events E2E Tests: Process Execute Failed",
+		Tags:        []string{"e2e", "instrumentation"},
+	}, nil
+}
+
+func (sig *e2eProcessExecuteFailed) GetSelectedEvents() ([]detect.SignatureEventSelector, error) {
+	return []detect.SignatureEventSelector{
+		{Source: "tracee", Name: "process_execute_failed"},
+		{Source: "tracee", Name: "init_namespaces"},
+	}, nil
+}
+
+func (sig *e2eProcessExecuteFailed) OnEvent(event protocol.Event) error {
+	eventObj, ok := event.Payload.(trace.Event)
+	if !ok {
+		return fmt.Errorf("failed to cast event's payload")
+	}
+
+	switch eventObj.EventName {
+	case "init_namespaces":
+		// The event does not support kernel versions 5.7 and older, so pass the test if run in such
```

Comment:
Was this the case for the event before? Or is the only way to fix the event breaking it for kernels older than 5.7? I'd be cautious introducing the fix if so.

Reply:
Previous symbols was not  reliable, any kernel could have missed it depending its compilation and configuration.
The new probes are an LSM function and sys_exit tail call, which should be reliable from kernel 5.8 on.
Supporting older kernels is very hard, as there are no real function there to sit on with access to the `bprm` object to extract from the execution information for the event.

Reply:
If the previous version doesn't give false positives, then I don't think it's responsible to break the event, and the fixed version should be supplied in addition (maybe we can add a simple kernel version condition to probe dependencies. could be a useful feature in general).
Otherwise, we should deprecate the event in versions which don't have the fix.

Reply:
To be honest, the best solution will be to run the current event if possible, as it is much simpler implementation.
In the case the symbol is missing, then we can use the new and more reliable solution.
However, this is not possible in the current architecture and the refactor is much too complicated for this PR as I see it.

Reply:
@NDStrahilevitz you can go over the new PR at #3881